### PR TITLE
feat: podman desktop extension

### PR DESCRIPTION
**Pull request by feloy**
An image can be created and pushed with `./scripts/build.sh <image-name> <tag>`

This creates a multi-platform image for (windows,linux,darwin) x (amd64,arm64).

Fixes https://github.com/redhat-developer/podman-desktop-image-checker-openshift-ext/issues/21

### CODE REVIEW

**Comment by lstocchi** [comment]
I built the extension, installed it on Desktop but i receive this message for every image.
image

**Comment by lstocchi** [comment]
Not sure if this is expected bc i have to do something or there is an error, so asking

**Comment by feloy** [comment]
> Not sure if this is expected bc i have to do something or there is an error, so asking

Which OS/arch are you using?

**Comment by lstocchi** [comment]
windows amd64

**Comment by feloy** [comment]
Which version of Podman desktop are you using? Particularly, does it include this patch? https://github.com/containers/podman-desktop/pull/5069

**Comment by lstocchi** [comment]
> Which version of Podman desktop are you using? Particularly, does it include this patch? [containers/podman-desktop#5069](https://github.com/containers/podman-desktop/pull/5069)

i was using my local main branch but probably it was not rebased, let me re-try

**Comment by lstocchi** [comment]
Ok it was a problem with my branch. Works fine, thanks @feloy

File: podman-desktop-extension/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  },
+  "scripts": {
+    "build": "vite build",
+    "test": "vitest run --coverage --passWithNoTests",
+    "test:watch": "vitest watch --coverage --passWithNoTests",
+    "watch": "tsc -w"
+  },
+  "dependencies": {},
+  "devDependencies": {
+    "@podman-desktop/api": "https://gitpkg.now.sh/containers/podman-desktop/packages/extension-api?main",
```
Comment (lstocchi) [inline podman-desktop-extension/package.json:24]: Why don't you use a version here?

**Review by lstocchi — APPROVED** [review]
LGTM

File: podman-desktop-extension/package.json (hunk 1 above)
Comment (feloy) [reply podman-desktop-extension/package.json:24]: I don't think the latest release (1.5.3) includes the necessary changes in the API

Comment (jeffmaury) [reply podman-desktop-extension/package.json:24]: next is the version to use

File: Containerfile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ARG PLATFORM_ARG
+
+LABEL org.opencontainers.image.title="OpenShift Checker" \
+        org.opencontainers.image.description="Analyze a Containerfile and highlight the directives and commands which could cause an unexpected behavior when running on an OpenShift cluster." \
+        org.opencontainers.image.vendor="Red Hat" \
+        io.podman-desktop.api.version=">= 1.5.3"
+
+COPY --from=extension-builder /extension /extension
+
+COPY --from=cli-builder /opt/app-root/src/bin/doa.cross.linux.${PLATFORM_ARG} /extension/doa.linux
```
Comment (jeffmaury) [inline Containerfile:64]: Why don't we have an image per os/arch with a single binary ?

File: podman-desktop-extension/src/extension.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import * as extensionApi from '@podman-desktop/api';
+
+// Activate the extension asynchronously
+export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
+  // Create a provider with an example name, ID and icon
+  const provider = extensionApi.provider.createProvider({
```
Comment (jeffmaury) [inline podman-desktop-extension/src/extension.ts:23]: Should be removed, not needed

**Review by jeffmaury — CHANGES_REQUESTED** [review]

File: Containerfile (hunk 2 above)
Comment (feloy) [reply Containerfile:64]: After some tests, it appears that, on Windows, the Linux image will be downloaded, not the Windows one (as the container runtime is Linux-based).

For me, the smallest granularity we can have is the platform.

File: podman-desktop-extension/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "openshift-checker",
+  "displayName": "OpenShift Checker",
```
Comment (jeffmaury) [inline podman-desktop-extension/package.json:3]:
```suggestion
  "displayName": "Red Hat OpenShift Checker",
```

**Review by jeffmaury — COMMENTED** [review]
I can't test the extension: here is what I have when I run it with --extension-folder

image

I tested against main and v1.6.3 of Podman Desktop source code

File: podman-desktop-extension/package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  "scripts": {
+    "build": "vite build",
+    "test": "vitest run --coverage --passWithNoTests",
+    "test:watch": "vitest watch --coverage --passWithNoTests",
+    "watch": "tsc -w"
+  },
+  "dependencies": {},
+  "devDependencies": {
+    "@podman-desktop/api": "next",
+    "@types/node": "^20.9.4",
```
[the remark this answers is not in the document]
Comment (feloy) [reply podman-desktop-extension/package.json:25]: Are you trying to run by moving the sources into the podman desktop sources and building it as the same time as PD? 

Or by downloading an image from the Extensions page?

[the remark this answers is not in the document]
Comment (jeffmaury) [reply podman-desktop-extension/package.json:25]: I'm using with different code bases but from the Podman Dekstop one, I'm running `yarn watch --extension-folder ../../redhat-developer/podman-desktop-image-checker-openshift-ext/podman-desktop-extension/`

[the remark this answers is not in the document]
Comment (feloy) [reply podman-desktop-extension/package.json:25]: OK, I didn't understand it was expected to be ran this way also.

I think I'll have to adapt the tsconfig.json file to make it compatible for both this way and built as an independent module into an image.

[the remark this answers is not in the document]
Comment (feloy) [reply podman-desktop-extension/package.json]: ok, I've made a fix to make it buildable with `yarn watch` and inside an image.

But with `yarn watch`, you still won't have the `doa` executable built, and it won't be found by the extension. Do we want to automate this in some way?

**Review by jeffmaury — CHANGES_REQUESTED** [review]
When I try to build the image, I have a warning that yarn build but there is no lock file

Also would be good to add node_modules to .gitignore

**Comment by feloy** [comment]
> When I try to build the image, I have a warning that yarn build but there is no lock file
> 
> Also would be good to add node_modules to .gitignore

It should be fixed with the latest commit

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**